astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,477 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Combine 3 images to produce a properly-scaled RGB image following Lupton et al. (2004).
+
+For details, see : http://adsabs.harvard.edu/abs/2004PASP..116..133L
+
+The three images must be aligned and have the same pixel scale and size.
+
+Example usage:
+    image_r = np.random.random((100,100))
+    image_g = np.random.random((100,100))
+    image_b = np.random.random((100,100))
+    image = lupton_rgb.make_lupton_rgb(image_r, image_g, image_b, filename='randoms.png')
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import numpy as np
+from . import ZScaleInterval
+
+
+__all__ = ['make_lupton_rgb']
+
+
+try:
+    import scipy.misc
+    scipy.misc.imresize  # checking if it exists
+    HAVE_SCIPY_MISC = True
+except (ImportError, AttributeError):
+    HAVE_SCIPY_MISC = False
+
+# NOTE: these methods would have come from LSST C++ code. They won't be available
+# in astropy until they are converted somehow.
+# from lsst.afw.display.displayLib import replaceSaturatedPixels, getZScale
+
+
+def compute_intensity(image_r, image_g=None, image_b=None):
+    """
+    Return a naive total intensity from the red, blue, and green intensities.
+
+    Parameters
+    ----------
+    image_r : `~numpy.ndarray`
+        Intensity of image to be mapped to red; or total intensity if ``image_g``
+        and ``image_b`` are None.
+    image_g : `~numpy.ndarray`, optional
+        Intensity of image to be mapped to green.
+    image_b : `~numpy.ndarray`, optional
+        Intensity of image to be mapped to blue.
+
+    Returns
+    -------
+    intensity : `~numpy.ndarray`
+        Total intensity from the red, blue and green intensities, or ``image_r``
+        if green and blue images are not provided.
+
+    Notes
+    -----
+    Inputs may be MaskedImages, Images, or numpy arrays and the return is
```

Comment:
I think this note referring to LSST objects "MaskedImages, Images" can just be removed. Here we use Numpy arrays and that's documented above.